packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,396 @@
+  });
+
+  it('renders an HTML input type=checkbox element', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    expect(fixture.nativeElement).toBeInTheDocument();
+  });
+
+  it('displays as CSS block element', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+
+    expect(fixture.nativeElement).toBeVisible();
+    expect(fixture.nativeElement).toHaveStyle({ display: 'block' });
+  });
+
+  it('renders a design system BEM block class name', async () => {
+    const { container } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    const checkbox = container.querySelector(':only-child');
+    expect(checkbox).toHaveClass('utrecht-checkbox');
+  });
+
+  describe('checked variant', () => {
+    it('is not checked by default', async () => {
+      await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).not.toBeChecked();
+    });
+
+    it('omits non-essential checked attributes when not checked', async () => {
+      await render('<input checked="false" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+      const checkbox = screen.getByRole('checkbox');
+
+      expect(checkbox).not.toHaveAttribute('aria-checked');
+      expect(checkbox).toHaveAttribute('checked');
+    });
+
+    it('can have a checked state', async () => {
+      const handleChange = jest.fn();
+      await render('<input (change)="sendValue($event)" [checked]="checked" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          sendValue: handleChange,
+          checked: true,
+        },
+      });
+
+      const checkbox = screen.getByRole('checkbox');
+      checkbox?.click();
+
+      expect(handleChange).toHaveBeenCalledTimes(1);
+      expect(checkbox).toBeChecked();
+    });
+
+    it('can have a checked state in CSS', async () => {
+      const { container } = await render('<input checked="true" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+      const checkbox = container.querySelector(':checked');
+      expect(checkbox).toBeInTheDocument();
+    });
+  });
+
+  describe('invalid variant', () => {
+    it('renders a design system BEM modifier class name', async () => {
+      const { container } = await render('<input [invalid]="invalid" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          invalid: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]');
```

Comment:
I suppose it would be more to the point to use `querySelector('.utrecht-checkbox')` in these BEM modifier class name tests.